src/main/java/stream/flarebot/flarebot/util/votes/VoteGroup.java

```diff
@@ -0,0 +1,119 @@
+package stream.flarebot.flarebot.util.votes;
+
+import net.dv8tion.jda.core.EmbedBuilder;
+import net.dv8tion.jda.core.entities.Message;
+import net.dv8tion.jda.core.entities.User;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VoteGroup {
+    private int yesVotes = 0;
+    private int noVotes = 0;
+
+    private List<Long> users = new ArrayList<>();
+    private List<Long> allowedUsers = null;
+
+    private String messageDesc;
+
+    private Message voteMessage;
+    private EmbedBuilder votesEmbed;
+
+    public VoteGroup(String messageDesc) {
+        this.messageDesc = messageDesc;
+    }
+
+    public boolean addVote(Vote vote, User user) {
+        if (allowedUsers != null && voteMessage != null) {
+            if (!allowedUsers.contains(user.getIdLong())) {
+                return false;
+            }
+        }
+        if (!users.contains(user.getIdLong())) {
+            if (vote == Vote.NO) {
+                noVotes++;
+            } else {
+                yesVotes++;
+            }
+            if(allowedUsers.size() == votes()) {
+                VoteUtil.finishNow(messageDesc, voteMessage.getGuild());
+                return true;
+            }
+            users.add(user.getIdLong());
+            if (voteMessage != null) {
+                votesEmbed.clearFields();
+                votesEmbed.addField("Yes Votes", String.valueOf(yesVotes), true);
+                votesEmbed.addField("No Votes", String.valueOf(noVotes), true);
+                voteMessage.editMessage(votesEmbed.build()).queue();
+            }
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
Just gonna save time and say remove all bodies where they can be, also chain the builder